DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -164,9 +193,17 @@ public override void Install()
                         bSuccess = this.InstallScriptFile(this.InstallScript);
                         installedVersion = this.InstallScript.Version;
                     }
-                }
-
-                // Then process remain Install/Upgrade Scripts
+                }
+                else

```

Comment:
I was thinking that `Upgrade` scripts didn't run on first install, but they do, so I think `PreUpgrade` script also run on first install.

Reply:
This is a problem IMHO. The upgrade scripts *should* IMHO adhere to their naming and run on upgrades, not install. From what I understand the use case to be, it would serve to prepare the existing data/schema for an upgrade. So in this case ripping out SPROCs, views etc. Obviously you would not want to run this on first install as there is nothing to rip out or you're ripping out stuff you just installed using the install script.

Reply:
I agree in principle, but that seems like a breaking change.  Do we want to hold this change until v10? /cc @mitchelsellers

Reply:
The preupgrade scripts are new, so there is no change here, only additional functionality